src/utils/use-color-theme.ts

```diff
@@ -12,52 +12,68 @@ export function getAvailableThemes(): Record<string, string> {
 
 const DEFAULT_THEME = "dark";
 
-function getPreference() {
+export function getColorThemePreference() {
   const pref = localStorage.getItem("color-scheme-preference");
   if (pref && getAvailableThemes()[pref]) return pref;
   return DEFAULT_THEME;
 }
 
-export function useColorTheme() {
-  const [pref, setPref] = useState(getPreference());
-
-  useEffect(() => {
-    const root = document.documentElement;
-    if (pref === "dark") {
-      root.classList.remove("theme-light", "theme-system");
-      root.classList.add("theme-dark");
-    } else if (pref === "light") {
-      root.classList.remove("theme-dark", "theme-system");
-      root.classList.add("theme-light");
-    } else {
-      root.classList.remove("theme-dark", "theme-light");
-      root.classList.add("theme-system");
-    }
-
-    localStorage.setItem("color-scheme-preference", pref);
-  }, [pref]);
+export function getColorThemePreferenceResolveSystem() {
+  const [currentTheme] = getColorThemePreference();
 
   const isDarkMode = useMedia("(prefers-color-scheme: dark)");
 
+  if (currentTheme === "system") {
+    return isDarkMode ? "dark" : "light";
+  }
+
+  return currentTheme;
+}
+
+function applyColorThemeNotPersistent(theme: string, isDarkMode: boolean) {
```

Comment:
I think this can just be `applyColorTheme` and then have a separate `persistColorTheme` function.